lib/axon/metrics.ex

```diff
@@ -46,21 +46,29 @@ defmodule Axon.Metrics do
         0.6666666865348816
       >
 
+      iex> Axon.Metrics.accuracy(Nx.tensor([[0, 1, 0], [1, 0, 0]]), Nx.tensor([[0, 1, 0], [0, 1, 0]]))
+      #Nx.Tensor<
+        f32
+        0.5
+      >
+
   """
   defn accuracy(y_true, y_pred) do
-    transform({y_true, y_pred}, fn {y_true, y_pred} ->
-      if elem(Nx.shape(y_pred), Nx.rank(y_pred) - 1) == 1 do
-        y_pred
-        |> Nx.greater(0.5)
-        |> Nx.equal(y_true)
-        |> Nx.mean()
-      else
-        y_true
-        |> Nx.argmax(axis: -1)
-        |> Nx.equal(Nx.argmax(y_pred, axis: -1))
-        |> Nx.mean()
-      end
-    end)
+    transform({y_true, y_pred}, &calculate_accuracy/1)
+  end
+
+  defp calculate_accuracy({y_true, y_pred}) do
+    if elem(Nx.shape(y_pred), Nx.rank(y_pred) - 1) == 1 do
```

Comment:
I think just this equality should be done inside transform, returning a "boolean" tensor. The rest of the calculation can be  done outside of transform

Reply:
Good point. I have added f0a4d64263d294ca0f90e38c05abf2489ea0e253 fixup. Let me know if there is a better way to write this.